tests/unit/task_system_test.c

```diff
@@ -373,39 +386,73 @@ CTEST2(task_system, test_task_system_creation_with_bg_threads)
 /*
  * ------------------------------------------------------------------------
  * Test creation of task system using up the threads for background threads.
- * Verify that the next user-thread creation should fail.
+ * Verify ththe we can create just one more user-thread and that the next
+ * user-thread creation should fail with a proper error message.
  * ------------------------------------------------------------------------
  */
-CTEST2(task_system, test_use_all_threads_for_bg_threads)
+CTEST2(task_system, test_use_all_but_one_threads_for_bg_threads)
```

Comment:
Modified this previously added test case to go the extra distance of showing that you can still create a user thread when there is one more slot free. And that the next thread creation will fail once we run out of the limit.